### DISCUSSION TITLE
INIT VALUE FROM EEPROM FOR ITEM_INT_RANGE(...)

### DISCUSSION THREAD

**Discussion by lasp53** [forntoh/LcdMenu#261] [Q&A]
Hello, please indicate how to set initial value from eeprom for the ITEM_INT_RANGE(...)
regards

**Comment by axa70** [comment]
facing the same problem but couldn't solve it

here is what I tried
```
//initial setpoint values
int IN_VAL_HU;

void setup(){

  heater_upper = EEPROM.read(ADD_HUP);
  if (heater_upper < 1 || heater_upper > 100) {
      heater_upper = D_VAL_HU;
      EEPROM.update(ADD_HUP, heater_upper);
      Serial.println("Heater upper default value updated");
  }
  IN_VAL_HU = (int)heater_upper;
  Serial.print("HUSP: "); Serial.println(IN_VAL_HU);

}

// submenu 2 for heater
MENU_SCREEN(heater_settings, heater_settings_items,
    ITEM_WIDGET(
        "H U",
        [](byte hu) {
            EEPROM.update(ADD_HUP, hu);  // Store the selected quantity in the global variable
        },
        WIDGET_RANGE(IN_VAL_HU, 1, 80, 100, "%d", 0)),
```  

So I am guessing the menu is rendered before the IN_VAL_HU is updated from the EEPROM.

**Reply by fvdbeek** [reply]
Indeed, MENU_SCREEN is a macro that is expanded at compile time. You should use the setter method of ITEM_WIDGET to update the value. I have given a simple example in [this discussion](https://github.com/forntoh/LcdMenu/discussions/309#discussioncomment-12205549).
